src/rogue/interfaces/memory/Transaction.cpp

```diff
@@ -139,6 +174,20 @@ void rim::Transaction::errorPy(std::string error) {
          type_,id_,address_,size_,error_.c_str());
 
    cond_.notify_all();
+   
+   // If applicable, notify parent transaction about completion of a sub-transaction 
+   if (isSubTransaction_)  {
+      // Get a shared_ptr to the parent transaction
+      rim::TransactionPtr parentTran = parentTransaction_.lock();
+      if (parentTran) {
+         // Remove own ID from parent subtransaction map
+        parentTran->subTranMap_.erase(id_);
+
+         // If this is the last sub-transaction, notify parent transaction it is all done
+         if (parentTran->subTranMap_.empty() and parentTran->doneCreatingSubTransactions_)
+           parentTran->error("Transaction error. Subtransaction failed.");
```

Comment:
I think we should check to see if an error already exists in the transaction and either append the error or don't replace the existing error?

Reply:
We do get a lock on the parent at line 181, no?